apps/prairielearn/src/pages/instructorCourseAdminInstances/instructorCourseAdminInstances.html.ts

```diff
@@ -225,3 +253,78 @@ function PopoverEndDate() {
     </p>
   `.toString();
 }
+
+function CreateCourseInstanceModal({
+  csrfToken,
+  initialStartDateFormatted,
+  initialEndDateFormatted,
+}: {
+  csrfToken: string;
+  initialStartDateFormatted?: string;
+  initialEndDateFormatted?: string;
+}) {
+  return Modal({
+    id: 'createCourseInstanceModal',
+    title: 'Create Course Instance',
+    formMethod: 'POST',
+    body: html`
+      <div class="form-group">
+        <label for="short_name">CIID (Course Instance Identifier)</label>
+        <input
+          type="text"
+          class="form-control"
+          id="short_name"
+          name="short_name"
+          required
+          pattern="[\\-A-Za-z0-9_\\/]+"
+          title="Please enter a valid format: only letters, numbers, hyphens, underscores, backslashes, and forward slashes."
+        />
+        <small class="form-text text-muted"
+          >The recommended format is <code>Fa19</code> or <code>Fall2019</code>. Use only letters,
+          numbers, dashes, and underscores, with no spaces.
+        </small>
+      </div>
+      <div class="form-group">
+        <label for="long_name">Long Name</label>
+        <input type="text" class="form-control" id="long_name" name="long_name" required />
+        <small class="form-text text-muted">
+          This is the full name of the course instance, such as "Fall 2019" or "Spring 2020".
+        </small>
+      </div>
+      <div class="form-group">
```

Comment:
For consideration: immediately forcing someone to pick when a course instance should be accessible to students might make them bounce if they just want to experiment with the platform. Maybe we should have a "Make course instance available to students" checkbox that will disable and hide the date inputs? cc @mfsilva22 I'd like your input here.

Reply:
I'll wait for @mfsilva22 's input prior to making this change.

Reply:
I agree that we should not force instructors to enter the dates at first. A checkbox is a good idea. I would be more inclined to not request for dates as default. We can add some help text saying the dates can be edited later.

Reply:
I added the "Make course instance available to students" checkbox:

<img width="1079" alt="image" src="https://github.com/user-attachments/assets/8767a0e8-746b-4f7a-8b61-fd78d69c6042" />

<img width="1078" alt="image" src="https://github.com/user-attachments/assets/c9e2d67b-8738-4d8f-a49e-7754681c1227" />

Reply:
Excellent 🙌